### DISCUSSION TITLE
Microcks APIs taking more time to respond nearly >1sec

### DISCUSSION THREAD

**Discussion by sunilg009** [microcks/microcks#945] [Community]
Hi Team,
I have mocked around 250+ apis in single collection in microcks and see that, latency on each apis take around 1sec.

ex:= we have 50 TestScenarios to be executed in a feature file.
each scenario calls around 15apis by service code, which is been mocked in microcks.
Here we have 50 * 15= 750Apis calls , which will be performed over microcks.
So Here we see that microcks response for each api is >1sec.
So our testExecution is taking more time.

Any Solution for this or any workaround , please suggest.

Thanks in Advance.

**Comment by yada** [comment]
@sunilg009, we will send you some troubleshooting points to verify/trying to identify where this latency is coming from.

But please follow up on it, see: 
https://github.com/microcks/microcks/issues/914
and 
https://github.com/microcks/microcks/issues/890

We are waiting for your feedback on the issues above ;-)

Last but not least, I am still waiting for your email (yacine AT microcks.io) to schedule a community call; please include Ritika Singla in the thread so we can do a single call with all. Thanks.

**Reply by sunilg009** [reply]
sure will try it out and update you in a while, THanks for the reply.

**Comment by lbroudoux** [comment]
Hi @sunilg009,

Just to be sure to understand: are you seeing slow response time on mock invocations or on test executions? From what I understand, I think you're talking about mock invocations, but correct me if I'm wrong.

Different things to check to investigate further:
* Do you also see this important latency on unit calls?
* Have you turned on `DEBUG` levels to get timestamps and check what's slow? Could be several things like request parsing, database connection, script dispatching, response formatting, event procesing...
* Have you checked the performance metrics available from the Prometheus endpoint? Check https://microcks.io/documentation/using/advanced/monitoring/#technical-metrics
* If you see this behavior only under heavy load, have you tried disabling the `mockInvocationStats`? This helps some other users in the community to have better throughput

Let us know your findings!
